src/views/DashboardView.ts

```diff
@@ -0,0 +1,85 @@
+import vscode, {
+  commands,
+  ProviderResult,
+  ThemeIcon,
+  TreeItem,
+  Uri,
+  Event,
+  TreeDataProvider,
+  TreeItemCollapsibleState,
+  TreeView,
+} from "vscode";
+import {DashboardCommands} from "../commands";
+import {Constants} from "../Constants";
+import {vscodeEnvironment} from "../helpers";
+
+class DashboardTreeItem extends TreeItem {
+  constructor(
+    readonly label: string,
+    readonly description: string,
+    readonly iconPath: string | Uri | {light: string | Uri; dark: string | Uri} | ThemeIcon,
+    readonly contextValue: string,
+    readonly children?: DashboardTreeItem[]
+  ) {
+    super(label, children === undefined ? TreeItemCollapsibleState.None : TreeItemCollapsibleState.Expanded);
+    this.description = description;
+    this.iconPath = iconPath;
+    this.contextValue = contextValue;
+  }
+}
+
+class DashboardTreeDataProvider implements TreeDataProvider<DashboardTreeItem> {
+  onDidChangeTreeData?: Event<void | DashboardTreeItem | DashboardTreeItem[] | null | undefined> | undefined;
+
+  parent: DashboardTreeItem[];
+
+  constructor() {
+    this.parent = [
+      new DashboardTreeItem(
+        Constants.treeItemData.service.dashboard.label,
+        `${Constants.networkProtocols.http}${Constants.localhost}:${Constants.dashboardPort}`,
+        Constants.treeItemData.service.dashboard.iconPath,
+        Constants.treeItemData.service.dashboard.contextValue
+      ),
+    ];
+  }
+
+  getTreeItem(element: DashboardTreeItem): TreeItem | Thenable<TreeItem> {
+    return element;
+  }
+
+  getChildren(element?: DashboardTreeItem | undefined): ProviderResult<DashboardTreeItem[]> {
+    if (element === undefined) {
+      return this.parent;
+    }
+    return element.children;
+  }
+}
+
+/**
+ * Register our dashboard view as:
+ *  viewID: "truffle-vscode.views.dashboard"
+ *
+ * @param viewId - the viewId - defaults to above.
+ */
+export function registerDashboardView(viewId: string = "truffle-vscode.views.dashboard"): TreeView<DashboardTreeItem> {
+  commands.registerCommand(`${viewId}.startDashboardServer`, async () => {
+    await DashboardCommands.startDashboardCmd();
+  });
+  commands.registerCommand(`${viewId}.stopDashboardServer`, async () => {
+    await DashboardCommands.stopDashboardCmd();
+  });
+  commands.registerCommand(`${viewId}.restartDashboardServer`, async () => {
+    await DashboardCommands.stopDashboardCmd().then(() => DashboardCommands.startDashboardCmd());
+  });
+  commands.registerCommand(`${viewId}.copyRPCEndpointAddress`, async () => {
+    await vscodeEnvironment.writeToClipboard(
+      `${Constants.networkProtocols.http}${Constants.localhost}:${Constants.dashboardPort}/rpc`
```

Comment:
Noticing that when selecting "Copy RPC Endpoint Address" it's appending a `/rpc` at the end of the URL that's copied which is invalid when attempting to view in the browser. Also it might be better to change it to "Open Dashboard" which opens the browser at the URL (which would making copying the endpoint redundant too)

Reply:
Hey @kevinbluer

I add another option called: "Open Dashboard in Web Browser", above the option "Copy RPC Endpoint Address". I've decided to keep the second one because sometimes we need the RPC to execute some tests using a client. Is that ok for you to have these 2 options?

<img width="341" alt="Screenshot 2022-07-04 at 11 47 39" src="https://user-images.githubusercontent.com/32708942/177139738-7d5b04af-96f0-4857-b80b-7416c7c51eb8.png">